### PR TITLE
feat(plugin): Create AWS Lambda Integration plugin

### PR DESCRIPTION
**Pull request by timerickson**
Implement #1005 - AWS Lamba Integration.
For design consideration/review only - not for merge consideration as yet.
Test cases to follow.
Create MVP Lambda plugin. See README.md for details.
Note all kong/plugins/aws-lambda/api-gateway and kong/plugins/aws-lambda/resty files are dependencies added to the repo - please see README.md for details.

The only contributed code is the handler.lua and shema.lua, along with the README.md.

This has been tested with a hello-world-py lamba in my account and it worked on my machine ;-)

### CODE REVIEW

**Comment by ahmadnassri** [comment]
nice work! the team will review and provide feedback.

File: kong/plugins/aws-lambda/api-gateway/aws/AWSBasicCredentials.lua [hunk 1]
```diff
@@ -0,0 +1,42 @@
+--[[
+  Copyright 2016 Adobe Systems Incorporated. All rights reserved.
```
Comment (tecnobrat) [inline kong/plugins/aws-lambda/api-gateway/aws/AWSBasicCredentials.lua]: Not sure how a contribution to an open source product can be Copyright Adobe :)

Comment (Tieske) [inline kong/plugins/aws-lambda/api-gateway/aws/AWSBasicCredentials.lua]: The copyright only states the owner of the copyright. The license itself actually tells you what you can do with it. As long as you do not violate the license all is ok. The tricky part is to make sure you don't violate it....

Comment (timerickson) [inline kong/plugins/aws-lambda/api-gateway/aws/AWSBasicCredentials.lua]: Agreed, @Tieske . [https://github.com/Mashape/kong/blob/bbaea4ca0a62b2050b0457af8787b98b48349a76/kong/vendor/classic.lua#L4](classic.lua, Copyright (c) 2014, rxi) is already exhibiting this. Also, all dependencies in the rockspec. Except the api-gateway files are Apache 2.0 license whereas the classic.lua is MIT.

Any feedback/suggestions about whether/how better to include this? Or is this a no-go? I understand and called out in the README.md that it seemed odd. This reminds me that this is not a purely open source project, much like adobe's project. I leave it to someone smarter/more entangled than myself to make that call. Rewriting that portion *is* something I'm interested in, but would take a while for obvious reasons, and would probably need to be dependent on other projects at some level for certain things like the crypto.

Comment (sinzone) [inline kong/plugins/aws-lambda/api-gateway/aws/AWSBasicCredentials.lua]: @timerickson we prefer to not include this for both legal and long term business reasons. We accept MIT or  better Apache 2.0.

Comment (Tieske) [inline kong/plugins/aws-lambda/api-gateway/aws/AWSBasicCredentials.lua]: Does the hmac and http code have to be included like this? aren't there resty modules for that?

The adobe code is Apache 2.0, haven't checked the license of the http.lua file by James Hurst, but looks like MIT.

Comment (timerickson) [inline kong/plugins/aws-lambda/api-gateway/aws/AWSBasicCredentials.lua]: @sinzone See new comments on #1005 for a new PR imminent which switches to a different non-adobe MIT-licensed lib.

**Comment by timerickson** [comment]
Closing in favor of #1190

**Pull request closed without merging**